internal/admission/handler.go

```diff
@@ -0,0 +1,310 @@
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	_, err = w.Write(data)
+	if err != nil {
+		h.Logger.WithError(err).Error("failed to write response")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+var (
+	consumerGVResource = metav1.GroupVersionResource{
+		Group:    configuration.SchemeGroupVersion.Group,
+		Version:  configuration.SchemeGroupVersion.Version,
+		Resource: "kongconsumers",
+	}
+	pluginGVResource = metav1.GroupVersionResource{
+		Group:    configuration.SchemeGroupVersion.Group,
+		Version:  configuration.SchemeGroupVersion.Version,
+		Resource: "kongplugins",
+	}
+	clusterPluginGVResource = metav1.GroupVersionResource{
+		Group:    configuration.SchemeGroupVersion.Group,
+		Version:  configuration.SchemeGroupVersion.Version,
+		Resource: "kongclusterplugins",
+	}
+	kongIngressGVResource = metav1.GroupVersionResource{
+		Group:    configuration.SchemeGroupVersion.Group,
+		Version:  configuration.SchemeGroupVersion.Version,
+		Resource: "kongingresses",
+	}
+	secretGVResource = metav1.GroupVersionResource{
+		Group:    corev1.SchemeGroupVersion.Group,
+		Version:  corev1.SchemeGroupVersion.Version,
+		Resource: "secrets",
+	}
+	gatewayGVResource = metav1.GroupVersionResource{
+		Group:    gatewayv1beta1.SchemeGroupVersion.Group,
+		Version:  gatewayv1beta1.SchemeGroupVersion.Version,
+		Resource: "gateways",
+	}
+	httprouteGVResource = metav1.GroupVersionResource{
+		Group:    gatewayv1beta1.SchemeGroupVersion.Group,
+		Version:  gatewayv1beta1.SchemeGroupVersion.Version,
+		Resource: "httproutes",
+	}
+)
+
+func (h RequestHandler) handleValidation(ctx context.Context, request admissionv1.AdmissionRequest) (
+	*admissionv1.AdmissionResponse, error,
+) {
+	responseBuilder := NewResponseBuilder(request.UID)
+
+	switch request.Resource {
+	case consumerGVResource:
+		return h.handleKongConsumer(ctx, request, responseBuilder)
+	case pluginGVResource:
+		return h.handleKongPlugin(ctx, request, responseBuilder)
+	case clusterPluginGVResource:
+		return h.handleKongClusterPlugin(ctx, request, responseBuilder)
+	case secretGVResource:
+		return h.handleSecret(ctx, request, responseBuilder)
+	case gatewayGVResource:
+		return h.handleGateway(ctx, request, responseBuilder)
+	case httprouteGVResource:
+		return h.handleHTTPRoute(ctx, request, responseBuilder)
+	case kongIngressGVResource:
+		return h.handleKongIngress(ctx, request, responseBuilder)
+	default:
+		return nil, fmt.Errorf("unknown resource type to validate: %s/%s %s",
+			request.Resource.Group, request.Resource.Version,
+			request.Resource.Resource)
+	}
+}
+
+func (h RequestHandler) handleKongConsumer(
+	ctx context.Context,
+	request admissionv1.AdmissionRequest,
+	responseBuilder *ResponseBuilder,
+) (*admissionv1.AdmissionResponse, error) {
+	consumer := configuration.KongConsumer{}
+	deserializer := codecs.UniversalDeserializer()
+	_, _, err := deserializer.Decode(request.Object.Raw,
+		nil, &consumer)
```

Comment:
Aren't we a bit too conservative with line length :) ?

Reply:
Yes, we are 🙂: a51abda